sunpy/map/mapbase.py

```diff
@@ -2158,19 +2161,26 @@ def draw_quadrangle(self, bottom_left, *, width: u.deg = None, height: u.deg = N
         """
         axes = self._check_axes(axes, allow_non_wcsaxes=False)
 
-        bottom_left, top_right = get_rectangle_coordinates(
-            bottom_left, top_right=top_right, width=width, height=height)
+        if isinstance(bottom_left, astropy.units.quantity.Quantity):
+            anchor, _, top_right, _ = self._parse_submap_quantity_input(bottom_left, top_right, width, height)
+            width, height = top_right - anchor
+            kwargs.update({"vertex_unit": u.pix})
+        else:
+            bottom_left, top_right = get_rectangle_coordinates(
+                bottom_left, top_right=top_right, width=width, height=height)
 
-        width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
-        height = top_right.spherical.lat - bottom_left.spherical.lat
+            width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
+            height = top_right.spherical.lat - bottom_left.spherical.lat
+            transform = axes.get_transform(bottom_left.frame.replicate_without_data())
+            anchor = self._get_lon_lat(bottom_left)
+            kwargs.update({"transform": transform})
 
         kwergs = {
-            "transform": axes.get_transform(bottom_left.frame.replicate_without_data()),
```

Comment:
`transform` will now be defined on both logic branches, so put `transform` back into this dictionary
```
            "transform": transform,
```